### PR TITLE
Not operator

### PR DESCRIPTION
**Pull request by MacondoExpress**
# Description

Includes the NOT operator, this will substitute in future all the `_NOT` filters.

## Complexity

Complexity:

Low

# Issue

Closes 

- #2509 

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [x] Documentation has been updated
- [x] TCK tests have been updated
- [x] Integration tests have been updated
- [x] Example applications have been updated
- [x] New files have copyright header
- [x] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-stage-2632.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

File: packages/graphql/src/schema/create-relationship-fields/index.ts [hunk 1]
```diff
@@ -668,11 +680,19 @@ function createRelationshipFields({
...
                 ...(edgeWhereAggregationInput ? { edge: edgeWhereAggregationInput } : {}),
             },
         });
 
+        const deprecates_NOT = {
+            name: "deprecated",
+            args: {
+                reason: "Negate filters will be deprecated from version 4.0.0, use the NOT operator to achieve the same behavior",
+            },
+        };
```
Comment (mjfwebb) [inline packages/graphql/src/schema/create-relationship-fields/index.ts:694]: I see this is declared in multiple places. Perhaps we can introduce a new file to import it from?

File: docs/modules/ROOT/pages/subscriptions/filtering.adoc [hunk 2]
```diff
@@ -167,10 +162,10 @@ NOTE: `Where` will only filter by existing properties right before deletion.
...
 === Combining operators
 
-All above-mentioned operators can be combined using the `AND`/`OR` operators. 
+All above-mentioned operators can be combined using the `AND`/`OR`/`NOT` operators. 
 They accept an array argument with items of the same format as the `where` argument, which means they can also be nested to form complex combinations.
 
-Say we are picky fans of comedy movies and we only accept ratings below 7 for movies released before the 2000's. 
+Say we dislike comedy movies and we only accept ratings below 7 for movies released before the 2000's. 
 As an exception we also like the movie "The Matrix". However, we do not like any of its sequels. 
 We could subscribe to any updates that we are interested in as follows:
```
Comment (mjfwebb) [inline docs/modules/ROOT/pages/subscriptions/filtering.adoc:170]: The wording of this contrived example doesn't really make sense anymore with these changes (the exception from before of The Matrix being a non-comedy movie is no longer relevant)

**Review by mjfwebb — COMMENTED** [review]
Really big fan of this PR. Just some small comments :)

File: docs/modules/ROOT/pages/subscriptions/filtering.adoc (hunk 2 above)
Comment (MacondoExpress) [reply docs/modules/ROOT/pages/subscriptions/filtering.adoc:170]: Good catch here!

**Review by darrellwarde — APPROVED** [review]
This looks good! I think it's worth noting that this is currently not as well tested as it will be when this transition is complete, because we have not changed all of the tests over which use the filters which will be deprecated in 4.0.0. However, I think it's clear that this stuff will all work as expected, so I'm not too worried.

**Review by darrellwarde — APPROVED** [review]
LGTM, let's get it in!!

**Comment by MacondoExpress** [comment]
🎉 🎉  Finally is out there!!

**Pull request merged**